src/debug/debugSession.ts

```diff
@@ -270,11 +270,33 @@ export class DebugSession implements IDebugSession {
         await this.startDebugging(cwd, sessionName, proxyResult.proxyDebugPort, proxyResult.proxyAppPort, async () => {
             proxyResult.proxyProcess.kill();
             if (appName) {
-                await this.cleanupResource(`deployment/${appName}`);
+                kubeChannel.showOutput("The debug session exited.");
+                await this.promptForCleanup(`deployment/${appName}`);
             }
         });
     }
 
+    private async promptForCleanup(resourceId: string): Promise<void> {
+        const autoCleanupFlag = vscode.workspace.getConfiguration("vs-kubernetes")["vs-kubernetes.autoCleanupOnDebugTerminate"];
+        if (autoCleanupFlag) {
+            return await this.cleanupResource(resourceId);
+        }
+        const answer = await vscode.window.showWarningMessage(`Resource ${resourceId} will continue running on the cluster.`, "Clean Up", "Always Clean Up");
+        if (answer === "Clean Up") {
+            return await this.cleanupResource(resourceId);
+        } else if (answer === "Always Clean Up") {
+            // The object returned by VS Code API is readonly, clone it first.
+            const oldConfig = vscode.workspace.getConfiguration("vs-kubernetes");
+            const newConfig = {};
+            for (const key of Object.keys(oldConfig)) {
+                newConfig[key] = oldConfig[key];
+            }
+            newConfig["vs-kubernetes.autoCleanupOnDebugTerminate"] = true;
+            await vscode.workspace.getConfiguration().update("vs-kubernetes", newConfig);

```

Comment:
I believe this will set this option for the workspace, not the user.  Is that desired?

Reply:
If we feel user level is better I have code in progress for that on another branch: https://github.com/itowlson/vscode-kubernetes-tools/blob/98131ec800c3170054f935e070aae3221007e3ac/src/components/installer/installer.ts#L118.  Feel free to borrow/adapt!

Reply:
Yes, it will be saved in `workspace settings.json`.  Use `update` api will save it in `Global` or `Workspace`, it's a single answer question. The ideal api is to check if workspace settings.json exists or not, if so, save in workspace, otherwise, global. But i didn't find such API. Let me check if there is other way to achieve this.

Reply:
Have a look at `inspect` - I think that does what you want.

Reply:
Thanks to point that out. Have changed the update policy to make sure it will take effect under any situation.
* Always update global settings
* If workspaceFolder value exists, update it
* If workspace value exists, update it